app/src/main/java/com/infomaniak/mail/ui/main/folder/ThreadListFragment.kt

```diff
@@ -521,6 +521,18 @@ class ThreadListFragment : TwoPaneFragment(), SwipeRefreshLayout.OnRefreshListen
 
             waitingBeforeNotifyAdapter = threadListViewModel.isRecoveringFinished
 
+            deletedItemsIndices = { indices ->
+                if (isMultiSelectOn) {
+                    val previousThreads = threadListAdapter.dataSet.filterIsInstance<Thread>()
+                    var shouldPublish = false
+                    indices.forEach {
+                        val isRemoved = mainViewModel.selectedThreads.remove(previousThreads[it])
+                        if (isRemoved) shouldPublish = true
+                    }
+                    if (shouldPublish) publishSelectedItems()
+                }
+            }
```

Comment:
You should extract this in its own function 
That would solve the sonar code smell also